R/namespace_linter.R

```diff
@@ -123,74 +123,55 @@ namespace_linter <- function(check_exports = TRUE, check_nonexports = TRUE) {
   })
 }
 
-get_all_exports <- function(namespace) {
-  c(getNamespaceExports(namespace), names(.getNamespaceInfo(namespace, "lazydata")))
+exported_symbols <- function(ns) c(getNamespaceExports(ns), names(.getNamespaceInfo(ns, "lazydata")))
+is_exported <- function(symbols, namespaces) {
+  mapply(
+    function(pkg_sym, pkg_ns) pkg_sym %in% exported_symbols(pkg_ns),
+    symbols, namespaces
+  )
 }
 
 build_ns_get_int_lints <- function(packages, symbols, symbol_nodes, namespaces, source_expression) {
-  lints <- list()
-
   ## Case 2: foo does not exist in pkg:::foo
-
-  non_symbols <- !vapply(
-    seq_along(symbols),
-    function(ii) symbols[[ii]] %in% ls(namespaces[[ii]], all.names = TRUE),
-    logical(1L)
+  non_symbols <- !mapply(
```

Comment:
Isn't `vapply()` faster and safer to use because of type checks?

Reply:
yes, though safety is less of a concern for internal functions since we always control the input.

basically it's a readability-efficiency tradeoff, carrying around the `[[ii]]` everywhere vs. the cleaner code with `mapply()`.

Reply:
changed back to `vapply()` now that all 3 loops are covered by one helper, makes the readability tax minimal.